internal/move2kubeapi/handlers/roles.go

```diff
@@ -0,0 +1,239 @@
+	}
+	logrus.Trace("HandleListRoles end")
+}
+
+func HandleCreateRole(w http.ResponseWriter, r *http.Request) {
+	logrus.Trace("HandleCreateRole start")
+	accessToken, err := common.GetAccesTokenFromAuthzHeader(r)
+	if err != nil {
+		logrus.Debugf("failed to get the access token from the request. Error: %q", err)
+		w.Header().Set(common.AUTHENTICATE_HEADER, common.AUTHENTICATE_HEADER_MSG)
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+	defer r.Body.Close()
+	bodyBytes, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		logrus.Debugf("failed to read the request body. Error: %q", err)
+		sendErrorJSON(w, "the request body is missing or incomplete", http.StatusBadRequest)
+		return
+	}
+	reqRole := types.Role{}
+	if err := json.Unmarshal(bodyBytes, &reqRole); err != nil {
+		logrus.Debug("failed to unmarshal the request body as json. Error:", err)
+		sendErrorJSON(w, "the request body is invalid.", http.StatusBadRequest)
+		return
+	}
+	logrus.Debug("trying to create the role:", reqRole)
+	timestamp, _, err := common.GetTimestamp()
+	if err != nil {
+		logrus.Errorf("failed to get the timestamp. Error: %q", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	reqRole.Id = uuid.New().String()
+	reqRole.Timestamp = timestamp
+	logrus.Debug("after generating a new id for the role:", reqRole)
+	authServerRole, err := reqRole.ToAuthServerRole()
+	if err != nil {
+		logrus.Errorf("failed to convert the request role into an authorization server role. Error: %q", err)
+		sendErrorJSON(w, "the role is invalid", http.StatusBadRequest)
+		return
+	}
+	logrus.Debug("after converting to auth server role:", authServerRole)
+	if _, err := common.AuthServerClient.CreateClientRole(context.TODO(), accessToken, common.Config.AuthServerRealm, common.Config.M2kClientIdNotClientId, authServerRole); err != nil {
+		logrus.Debug("failed to create the role at the authorization server. Error:", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	result := map[string]string{"id": reqRole.Id}
+	resultBytes, err := json.Marshal(result)
+	if err != nil {
+		logrus.Debug("Error:", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	w.Header().Set("Location", path.Clean(r.URL.Path+"/"+reqRole.Id))
+	w.Header().Set(common.CONTENT_TYPE_HEADER, common.CONTENT_TYPE_JSON)
+	w.WriteHeader(http.StatusCreated)
+	if _, err := w.Write(resultBytes); err != nil {
+		logrus.Errorf("failed to write the response body. Error: %q", err)
+	}
+	logrus.Trace("HandleCreateRole end")
+}
+
+func HandleReadRole(w http.ResponseWriter, r *http.Request) {
+	logrus.Trace("HandleReadRole start")
+	accessToken, err := common.GetAccesTokenFromAuthzHeader(r)
+	if err != nil {
+		logrus.Debugf("failed to get the access token from the request. Error: %q", err)
+		w.Header().Set(common.AUTHENTICATE_HEADER, common.AUTHENTICATE_HEADER_MSG)
+		w.WriteHeader(http.StatusUnauthorized)
+		return
+	}
+	roleId := mux.Vars(r)["role-id"]
```

Comment:
We seem to be using role-id in more than one place, better to use a const to make sure they don't end up diverging.

Reply:
done